packages/flutter/test/material/expansion_panel_test.dart

```diff
@@ -1521,4 +1527,104 @@ void main() {
     expect((mergeableMaterial.children.first as MaterialSlice).color, firstPanelColor);
     expect((mergeableMaterial.children.last as MaterialSlice).color, secondPanelColor);
   });
+
+  testWidgets('ExpansionPanelList expanded color test', (WidgetTester tester) async {
+    const Color expandedColor = Colors.orange;
+    const Color color = Colors.red;
+
+    const Key firstPanelKey = Key('firstPanelKey');
+    const Key secondPanelKey = Key('secondPanelKey');
+
+    await tester.pumpWidget(
+      const MaterialApp(
+        home: SingleChildScrollView(
+          child: SimpleExpansionPanelListTestWidget(
+            firstPanelKey: firstPanelKey,
+            secondPanelKey: secondPanelKey,
+            color: color,
+            expandedColor: expandedColor,
+          ),
+        ),
+      ),
+    );
+
+    final Finder finder = find.descendant(
+      of: find.ancestor(
+        of: find.byKey(firstPanelKey),
+        matching: find.byType(Row),
+      ),
+      matching: find.byType(IconButton),
+    );
+
+    IconButton iconButton = tester.widget(finder);
+
+    expect(iconButton.color, color);
+
+    await tester.tap(find.byType(ExpandIcon).at(0));
+    await tester.pumpAndSettle();
+
+    iconButton = tester.widget(find.byType(IconButton).at(0));
+
+    expect(iconButton.color, expandedColor);
+  });
+
+  testWidgets('ExpansionPanelRadio expanded color test', (WidgetTester tester) async {
+    const Color expandedColor = Colors.orange;
+    const Color color = Colors.red;
+
+    const Key firstPanelKey = Key('firstPanelKey');
+    const Key secondPanelKey = Key('secondPanelKey');
+
+    final List<ExpansionPanel> _demoItemsRadio = <ExpansionPanelRadio>[
+      ExpansionPanelRadio(
+        headerBuilder: (BuildContext context, bool isExpanded) {
+          return Text(isExpanded ? 'B' : 'A', key: firstPanelKey);
+        },
+        body: const SizedBox(height: 100.0),
+        value: 0,
+        canTapOnHeader: true,
+      ),
+      ExpansionPanelRadio(
+        headerBuilder: (BuildContext context, bool isExpanded) {
+          return Text(isExpanded ? 'D' : 'C', key: secondPanelKey);
+        },
+        body: const SizedBox(height: 100.0),
+        value: 1,
+        canTapOnHeader: true,
+      ),
+    ];
```

Comment:
1. Since you are tapping the `ExpandIcon` anyway, `canTapOnHeader` can be left unassigned (it'll default to false).
2. Since we don't need two items for testing this, a single one would suffice.
3. Since this test doesn't care about the header widget, returning a `const Text('A')` in `headerBuilder` would be better.